lib/Parse/ParseType.cpp

```diff
@@ -368,7 +368,8 @@ ParserResult<TypeRepr> Parser::parseType(Diag<> MessageID,
   // Parse an async specifier.
   SourceLoc asyncLoc;
   if (shouldParseExperimentalConcurrency() &&
-      Tok.isContextualKeyword("async")) {
+      Tok.isContextualKeyword("async")
+      && peekToken().isAny(tok::arrow, tok::kw_throws)) {
```

Comment:
```suggestion
      Tok.isContextualKeyword("async") && 
      peekToken().isAny(tok::arrow, tok::kw_throws)) {
```
To align with the previous `&&`

Reply:
- [ ] Align ampersands

I'll fix this with the fix for handling `func f() -> Int async throws { }`.